tests/utils_test.py

```diff
@@ -84,6 +84,30 @@ def test_maybe_get_index_polars() -> None:
     assert result is None
 
 
+def test_maybe_reset_index_pandas() -> None:
+    pandas_df = nw.from_native(
+        pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, index=[7, 8, 9])
+    )
+    result = nw.maybe_reset_index(pandas_df)
+    expected = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, index=[0, 1, 2])
+    assert_frame_equal(nw.to_native(result), expected)
+    pandas_series = nw.from_native(
+        pd.Series([1, 2, 3], index=[7, 8, 9]), series_only=True
+    )
+    result_s = nw.maybe_reset_index(pandas_series)
+    expected_s = pd.Series([1, 2, 3], index=[0, 1, 2])
+    assert_series_equal(nw.to_native(result_s), expected_s)
+
+
+def test_maybe_reset_index_polars() -> None:
+    df = nw.from_native(pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}))
+    result = nw.maybe_reset_index(df)
+    assert result is df
+    series = nw.from_native(pl.Series([1, 2, 3]), series_only=True)
+    result = nw.maybe_reset_index(series)
+    assert result is series
```

Comment:
```suggestion
    result_s = nw.maybe_reset_index(series)
    assert result_s is series
```

Reply:
Thanks for fixing this @MarcoGorelli! I wasn't sure why `mypy` in CI  was failing there